mcpserver/server.go

```diff
@@ -0,0 +1,465 @@
+// Copyright (c) 2023-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package mcpserver
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/mark3labs/mcp-go/mcp"
+	"github.com/mark3labs/mcp-go/server"
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/shared/mlog"
+)
+
+// MattermostMCPServer provides a high-level interface for creating an MCP server
+// with Mattermost-specific tools and authentication
+type MattermostMCPServer struct {
+	mcpServer    *server.MCPServer
+	authProvider AuthenticationProvider
+	logger       mlog.LoggerIFace
+	config       Config
+}
+
+// NewMattermostMCPServer creates a new Mattermost MCP server with the specified configuration
+func NewMattermostMCPServer(config Config, authProvider AuthenticationProvider, logger mlog.LoggerIFace) (*MattermostMCPServer, error) {
```

Comment:
Maybe we should pass in the transport here too.

Reply:
I decided to make separate constructors for each server type; stdio and oauth. I think it's a cleaner approach